lib/modules/datasource/docker/index.ts

```diff
@@ -628,12 +628,30 @@ export class DockerDatasource extends Datasource {
     currentDigest: string
   ): Promise<string | null | undefined> {
     try {
-      const manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        currentDigest,
-        'head'
-      );
+      let manifestResponse: HttpResponse<string> | null;
+
+      try {
+        manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentDigest,
+          'head'
+        );
+      } catch (_err) {
+        const err = _err instanceof ExternalHostError ? _err.err : _err;
+
+        if (
+          typeof err.statusCode === 'number' &&
+          err.statusCode >= 500 &&
+          err.statusCode < 600
+        ) {
+          // querying the digest manifest for a non existent image leads to a 500 statusCode
```

Comment:
you don't get a 404? 🤔 maybe some registries return a 404, so we should at least text it too.

Reply:
I did my tests with the [official docker registry](https://docs.docker.com/registry/) by following the instructions in [renovate-reproduction-18206](https://github.com/renovate-reproductions/18206). But even if another registry returns a 404 status code it would lead to the same outcome because then [`getManifestResponse` returns `null`](https://github.com/kegato/renovate/blob/5045146d900fde68133c8448a873f06407cb5daf/lib/modules/datasource/docker/index.ts#L448) which causes [this function to return `null`](https://github.com/kegato/renovate/blob/5045146d900fde68133c8448a873f06407cb5daf/lib/modules/datasource/docker/index.ts#L662). My change just adds another special case handling in this specific context.